tests/cluster/tests/22-replica-in-sync.tcl

```diff
@@ -69,28 +78,34 @@ test "Replica in loading state is hidden" {
     }
     R $master_id exec
 
-    # Check that replica started loading
-    wait_for_condition 1000 50 {
-        [s $replica_id loading] eq 1
+    # The master will be the last to know the replica
+    # is loading, so we will wait on that and assert
+    # the replica is loading afterwards. 
+    wait_for_condition 100 50 {
+        ![is_in_slots $master_id $replica]
     } else {
-        fail "Replica didn't enter loading state"
+        fail "Replica was always present in cluster slots"
     }
-    # Check that replica is not in cluster slots
-    assert {![is_in_slots $master_id $replica]}
+    assert_equal 1 [s $replica_id loading]
 
-    # Wait for sync to finish
-    wait_for_condition 1000 50 {
-        [s $replica_id loading] eq 0
+    # Check that replica finished loading
+    wait_for_condition 100 50 {
+        [s $replica_id master_sync_in_progress] eq 0
```

Comment:
isn't it the other way around?
i.e. first `master_sync_in_progress` goes back to 0, and only then `loading` starts.